packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -59,21 +68,68 @@ export class ConnectionStateHandler extends EventEmitterWithErrorHandling<IConne
         private readonly logger: ITelemetryLogger,
     ) {
         super();
+        this.prevClientLeftTimer = new Timer(
+            // Default is 90 sec for which we are going to wait for its own "leave" message.
+            this.handler.maxClientLeaveWaitTime ?? 90000,
+            () => {
+                this.leaveReceivedResult = false;
+                this.applyForConnectedState();
+            },
+        );
     }
 
-    public receivedAddMemberEvent(clientId: string, quorum: Quorum) {
+    // This is true when this client submitted any ops.
+    public clientSentOps(connectionMode: ConnectionMode) {
+        if (this._connectionState === ConnectionState.Connected) {
```

Comment:
I think it could be an assert, not IF statement, as we should only see ops sent when we are in connected mode (this includes "read" mode)